File: matlab/simpleAcq.m
%==================================================================================================
% simpleAcq.m - A simple Matlab script for ultrasound data acquisition with the ARIUS platform.
%--------------------------------------------------------------------------------------------------
% Copyright (c) 2019 us4us Ltd. / MIT License
%==================================================================================================
%% Setup the Arius platform.
json = fileread('PA-BFR.json'); % load TX/RX JSON config file

hal = DummyHAL(reshape(1:16, 2, 2, 2, 2)); 
% DummyHAL can take one optional argument 'data':
% - 3-D matrix with dimensions: (x, y, z) - single 'RF frame'
% - 4-D matrix with dimensions: (x, y, z, n) - collection of RF frames to cycle 
%   through; the last dimension corresponds to the RF frame number. 
%  When no argument is provided: a two random RF frames (32, 512, 32) are 
%  generated.

hal.configure(json);    % init HAL with the TX/RX JSON
hal.start();            % start TX/RX

[B,A] = butter(5, 1/25, 'high');    % define a RF high-pass filter

%% Acquire ultrasound data Frame-by-Frame
for i = 1:3
    signal = single(hal.getData());
    hal.sync();
    %figure(1);     % show raw RF data
    %signal_filt = filter(B, A, signal, [], 2); % filter raw RF with HP filter

    plot(signal(:,:)'); % plot RF signal
    %drawnow;
    %pause(0.1);
    disp(signal);
end

File: python/simpleAcq.py
#!/usr/bin/python3
# =============================================================================
# simpleAcq.py - a simple Python script for ultrasound data acquisition
# with the ARIUS device.
# -----------------------------------------------------------------------------
# Copyright (c) 2019 us4us Ltd. / MIT License
# =============================================================================
import arius as ar
import argparse
import numpy as np
import matplotlib.pyplot as plt
import time


if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="ARIUS example - simple acq.")
    parser.add_argument("--cfg", dest="cfg",
                        help="Path to the json configuration file.",
                        required=True)
    parser.add_argument("--data", dest="data",
                        help="Path to the RF data to load.",
                        required=False,
                        default=None
                        )
    args = parser.parse_args()

    plt.ion()

    print("Detected devices:")
    for device in ar.getAvailableDevices():
        print("--- %s" % (device.name))

    # Load the input data.
    data = None
    if args.data:
        data = np.load(args.data)
    hal = ar.getHALInstance('MOCKHAL', data=data)
    # Setup the platform.
    with open(args.cfg, "r") as f:
        hal.configure(f.read())
        pass

    # Start TX/RX.
    hal.start()
    # Perform data acquisition and display.

    fig, ax, line = None, None, None # matplotlib objects.
    for i in range(10):
        data, metadata = hal.getData()

        # Plot acquired i-th line.
        if i == 0:
            fig, ax = plt.subplots()
            line, = ax.plot(data[0, :, i])
        else:
            line.set_ydata(data[0, :, i])
            fig.canvas.draw()
            fig.canvas.flush_events()
        time.sleep(0.5)

        hal.sync(metadata.frameIdx)
    # Stop TX/RX.
    hal.stop()
